pubber/src/main/java/daq/pubber/BasicPoint.java

```diff
@@ -36,7 +36,11 @@ public BasicPoint(String name, boolean writable, String units) {
     dirty = true;
   }
 
-  abstract Object getValue();
+  protected abstract Object getValue() throws IllegalStateException;
```

Comment:
NB: I generally try to avoid the Java-explicit-exceptions model because, well, I think it's dumb.  You'll see the general style in the code is to try/catch all exceptions locally and re-throw as a RuntimeException with additional context.  So, unless there's a strong reason for it (e.g. something is actually using the IllegalStateException for some specific behavior), I'd prefer to keep it that way (so not have the `throws IllegalStateException` addition to the signature).